src/components/TodosFilter/TodosFilter.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[],
+  setTodos: (a: Todo[]) => void;
+  allTodos: Todo[];
+};
+
+export const TodosFilter: React.FC<Props> = ({ todos, setTodos, allTodos }) => {
+  enum Status {
+    all = 'all',
+    active = 'active',
+    completed = 'Completed',
+  }
+
+  const [filterBy, setFilterBy] = useState<Status>(Status.all);
+
+  const getFilteredTodos = async (filter: Status) => {
+    let filteredTodos: Todo[] = [];
+
+    switch (filter) {
+      case Status.active:
+        filteredTodos = allTodos.filter(todo => !todo.completed);
+        break;
+      case Status.completed:
+        filteredTodos = allTodos.filter(todo => todo.completed);
+        break;
+      default:
+        filteredTodos = allTodos;
+        break;
+    }
+
+    setTodos(filteredTodos);
+  };
```

Comment:
better to lift up filterBy state to App component and return filteredTodos from filter funciton. 
update state is redundant, async is redundant
filteredTodos also move to App component. it avoid making two state for todos.

```suggestion
  const [filterBy, setFilterBy] = useState<Status>(Status.all);

  const filteredTodos = (filter: Status) => {
    switch (filter) {
      case Status.active:
        return allTodos.filter(todo => !todo.completed);
        break;
      case Status.completed:
        return allTodos.filter(todo => todo.completed);
        break;
      default:
       return allTodos;
        break;
    }
  };
```